Slack documentation (#96)
kyma-incubator/hack-showcase: docs/slack-connector/README.md

# Slack Connector Installation<!-- omit in toc -->

- [Overview](#overview)
- [Installation in Kyma with Helm](#installation-in-kyma-with-helm)
  - [Prerequisites](#prerequisites)
  - [Steps](#steps)

## Overview

> Slack Connector is a component which allows contact from inside of Kyma environment to the Slack API.

Reviewer: ```suggestion
Slack Connector is a component which allows contact from the inside of the Kyma environment to the Slack API.
```
How about:
> ... which allows the Kyma environment to communicate with the Slack API. 

or 
> ... which allows for communication between the Kyma environment and the Slack API. 

?

## Installation in Kyma with Helm

### Prerequisites

- Connection to Kyma cluster
- Slack Connector Docker image

### Steps

> 1. Visit our [authentication page](https://auth-slack.herokuapp.com/). Click `Add to Slack`. You will be redirected to another page. Select your desired workspace and click `Allow`

Reviewer: ```suggestion
1. Go to the [authentication page](https://auth-slack.herokuapp.com/). Click `Add to Slack`. This redirects you to another page. Select your desired workspace and click `Allow`.
```
- Try to avoid using possessive pronouns like "our", "my", etc.
- Don't use future tense (_will_), use present tense.
- Don't use passive voice (_you are redirected_), use active voice (_this redirects you_).
- Punctuate your sentences. Remember to add periods at the end of your sentences. If you use sentences in a list, end them with periods.

>     >**NOTE:** If the link doesn't work it means that you have to create an application like that yourself. The tutorial for that can be found in the Slack API [documentation](https://api.slack.com/docs/oauth#flow).

Reviewer: ```suggestion
    >**NOTE:** If the link does not work, it means that you have to create an application like that yourself. To create an application, see [this tutorial](https://api.slack.com/docs/oauth#flow) in the Slack API documentation.
```
- Don't use contractions, use full words (_do not_ instead of _don't_). 
- I'm not sure what you mean by 
> ... an application like that

I think you need to be more specific. For example:
> If the link does not work, it means that an application that [does something - specify what it is it does] does not exist and you have to create it yourself. 

or 

>  If the link does not work, it means that there is no application that [does something - specify what it is it does].

and then:
> To create an application [that does X], see ...

2. Copy the authentication token. You will need it later in the Helm command.

> 3. Go to [Kyma repository](https://github.com/kyma-project/kyma) and run script `/installation/scripts/tiller-tls.sh` to get certificates needed for using Helm commands. By default they are stored in `~/.helm` directory. After that add `--tls` flag to every Helm command to authorize and authenticate yourself

Reviewer: ```suggestion
3. Go to [Kyma repository](https://github.com/kyma-project/kyma) and run the script `/installation/scripts/tiller-tls.sh` to get certificates needed for using Helm commands. By default they are stored in the `~/.helm` directory. After that, add `--tls` flag to every Helm command to authorize and authenticate yourself.
```
I'd consider changing `yourself` to `user` or something similar. Is it the user that you authenticate or an app or what? I think you need to be more specific and "yourself" is too colloquial.

> 4. Go to `chart/slackconnector` directory. Run the command to install Slack Connector:

Reviewer: ```suggestion
4. Go to the `chart/slackconnector` directory. Run this command to install Slack Connector:
```

    ``` shell
    helm install --set container.image={DOCKER_IMAGE} --set kymaAddress={KYMA_ADDRESS} --set slackBotToken={SLACK_TOKEN} -n {RELEASE_NAME} . --tls
    ```

>     >**CAUTION:** Kyma address should be in the right format. It must consist of domain name, without dot  character at the beggining, for example `35.187.32.214.xip.io`

Reviewer: ```suggestion
    >**CAUTION:** Make sure that the Kyma address is in the correct format. It must consist of the domain name, without the dot character at the beginning. For example, `35.187.32.214.xip.io`.
```
`It must` or `it consists`?
- Don't use `should`.
- Don't use `for example` in the middle of the sentence.
- I'd rephrase that:
> It must consist of the domain name, **without the dot character at the beginning**.

to something like:
>  It consists of the domain name and omits the dot at the beginning. 

if the dot is there in other cases and just here needs to be omitted.
